thoth/storages/graph/dgraph.py

```diff
@@ -693,6 +693,9 @@ def get_all_versions_python_package(
         if index_url:
             q = q + ' AND eq(index_url, "%s")' % index_url
 
+        # required fields
+        q += " AND has(index_url) AND has(package_name)"
```

Comment:
Is `package_name` required here? Can we parametrize injecting `has(index_url)` so we can query those which have no index assigned?

Reply:
The solution is to use @cascade so it will discard nodes which does not have both the attributes you are requesting. But I would suggest in that case, to have a parameter to add @cascade, because saying we want all versions of python packages we admit packages with no index. Therefore we can have ```only_thoth_known_index: str = False``` parameter in the query, so we don't modify previous logic, and we add @cascade which is inserted only if that parameter is set to True in the query. What do you think @fridex ?

Reply:
This should also solve https://github.com/thoth-station/adviser/issues/303 because there won't be that case in adviser. Sorry for the long name of the parameter :(